src/code/z_schedule.c

```diff
@@ -0,0 +1,282 @@
+#include "global.h"
+
+#define SCHEDULE_CALC_TIME(hour, minute, dest, temp) \
+    (temp) = (hour)*60.0f;                           \
+    (temp) += (minute);                              \
+    (dest) = (temp) * (0x10000 / 60 / 24.0f);        \
+    (dest) = SCHEDULE_CONVERT_TIME(dest);
+
+s32 Schedule_CheckFlagS(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    ScheduleCmdCheckFlagS* cmd = (ScheduleCmdCheckFlagS*)*script;
+    u16 flag = (cmd->flagByte << 8) | cmd->flagMask;
+
+    if (gSaveContext.weekEventReg[flag >> 8] & (flag & 0xFF)) {
+        *script = *script + cmd->offset;
+    }
+
+    return false;
+}
+
+s32 Schedule_CheckFlagL(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    ScheduleCmdCheckFlagL* cmd = (ScheduleCmdCheckFlagL*)*script;
+    u16 flag = (cmd->flagByte << 8) | cmd->flagMask;
+
+    if (gSaveContext.weekEventReg[flag >> 8] & (flag & 0xFF)) {
+        *script = *script + (s16)((cmd->offsetH << 8) | cmd->offsetL);
+    }
+
+    return false;
+}
+
+s32 Schedule_CheckTimeRangeS(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    s32 inRange = 0;
```

Comment:
```suggestion
    s32 inRange = false;
```
As you have below. You can also change all other uses of this temp to use `true` as well